cs-ssh/src/main/java/io/cloudslang/content/ssh/utils/IdentityKeyUtils.java

```diff
@@ -0,0 +1,97 @@
+package io.cloudslang.content.ssh.utils;
+
+
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import io.cloudslang.content.ssh.entities.IdentityKey;
+import io.cloudslang.content.ssh.entities.KeyData;
+import io.cloudslang.content.ssh.entities.KeyFile;
+import io.cloudslang.content.ssh.exceptions.SSHException;
+
+/**
+ * User: sacalosb
+ * Date: 07.01.2016
+ */
+public class IdentityKeyUtils {
+    private static final String KEY_HEADER = "-----BEGIN RSA PRIVATE KEY-----";
+    private static final String KEY_FOOTER = "-----END RSA PRIVATE KEY-----";
+    private static final String TYPE_HEADER = "Proc-Type: ";
+    private static final String DEK_HEADER = "DEK-Info: ";
+    private static final String NEW_LINE = "\n";
+
+    public static IdentityKey getIdentityKey(String privateKeyFile, String privateKeyString, String privateKeyPassphrase) {
+        IdentityKey identityKey = null;
+        if (StringUtils.isNotEmpty(privateKeyFile)) {
+            identityKey = new KeyFile(privateKeyFile);
+        }
+        if (StringUtils.isNotEmpty(privateKeyString)) {
+            if (identityKey != null) {
+                throw new IllegalArgumentException(String.format("%s and %s inputs are mutually exclusive. They can't be both set at the same time.",
+                        Constants.PRIVATE_KEY_FILE, Constants.PRIVATE_KEY_DATA));
+            }
+            identityKey = new KeyData(privateKeyString);
+        }
+        if (identityKey != null && StringUtils.isNotEmpty(privateKeyPassphrase)) {
+            identityKey.setPassPhrase(privateKeyPassphrase);
+        }
+        return identityKey;
+    }
+
+    public static void setIdentity(JSch jsch, IdentityKey identityKey) throws SSHException {
+        byte[] passPhrase = identityKey.getPassPhrase();
+        if (identityKey instanceof KeyFile) {
+            try {
+                jsch.addIdentity(((KeyFile) identityKey).getKeyFilePath(), passPhrase);
+            } catch (JSchException e) {
+                throw new SSHException("The keyFilePath is invalid.", e);
+            }
+        } else {
+            try {
+                jsch.addIdentity(((KeyData) identityKey).getKeyName(), ((KeyData) identityKey).getPrvKeyData(), null, passPhrase);
+            } catch (JSchException e) {
+                throw new SSHException("The " + Constants.PRIVATE_KEY_DATA + " is invalid.", e);
+            }
+        }
+    }
+
+    /*
+    This method was added to fix a defect from Studio.
+    ALM#29569 Studio doesn’t support a multi lined sensitive data input
```

Comment:
probably all this code is not needed that transforms the key, but it cannot hurt